### ISSUE TITLE
502 error in version 0.8.2

### ISSUE THREAD

**Issue by Aephir** [hassio-addons/addon-jupyterlab#324]
# Problem/Motivation
Can't open the web UI.

## Expected behavior
Opens web UI

## Actual behavior

Gives a 502 error (somewhat like #322, but this was supposedly addressed in v0.8.2)

## Steps to reproduce

Start the add-on (v0.8.2). Try to open web UI from add-on `info` tab.
I had it installed at previous version that did not work, but I also tried uninstalling and reinstalling v0.8.2. No change in behavior.

## Logs

```
[I 2022-09-15 21:02:50.579 ServerApp] Jupyter Server 1.18.1 is running at:
[I 2022-09-15 21:02:50.579 ServerApp] http://localhost:8888/lab?token=REDACTED
[I 2022-09-15 21:02:50.579 ServerApp]  or http://127.0.0.1:8888/lab?token=REDACTED
[I 2022-09-15 21:02:50.580 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 2022-09-15 21:02:50.587 ServerApp] No web browser found: could not locate runnable browser.
[C 2022-09-15 21:02:50.587 ServerApp] 
    
    To access the server, open this file in a browser:
        file:///root/.local/share/jupyter/runtime/jpserver-237-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/lab?token=REDACTED
     or http://127.0.0.1:8888/lab?token=REDACTED
[21:03:45] INFO: Starting NGinx...
nginx: [alert] could not open error log file: open() "/var/log/nginx/error.log" failed (2: No such file or directory)
```

**Comment by chancancode** [comment]
So I think https://github.com/hassio-addons/addon-jupyterlab/pull/321 is an incomplete fix. There is an underlying issue that caused the config file to be ineffective – note that previously the server was happily running under the root user with `c.NotebookApp.allow_root = True`. Adding the command line flag worked around that one issue, but the underlying root cause affects everything else in the config file too.

So that needs to be investigated/addressed, and I think if that can be resolved, we should be able to revert #321 as well.

**Comment by tjorim** [comment]
Thanks for your idea about jupyterlab ignoring the config file alltogether. I spent some time looking at the debug logs and checking what changed. Apparently some migration needed to happen to that file with the upgrade to jupyterlab 3. I opened  a PR addressing the issue but I still have no clue why it only triggered now (just like Frenck testing the previous release did not encounter the allow-root issue).
